From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nick Cao <nickcao@nichi.co>
Date: Tue, 10 Oct 2023 10:17:00 -0400
Subject: [PATCH] qtbase: check in the QML folder of this library does actually
 exist

In a modularized installation, this folder will be the location where
`qtbase` itself is installed, but `qtbase` does not have any QML
code, and `qmlimportscanner` will complain that it does not exist.
---
 src/tools/macdeployqt/shared/shared.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/tools/macdeployqt/shared/shared.cpp b/src/tools/macdeployqt/shared/shared.cpp
index 2b972a76c49..96c61b3824a 100644
--- a/src/tools/macdeployqt/shared/shared.cpp
+++ b/src/tools/macdeployqt/shared/shared.cpp
@@ -1300,9 +1300,12 @@ bool deployQmlImports(const QString &appBundlePath, DeploymentInfo deploymentInf
     }
     for (const QString &importPath : qmlImportPaths)
         argumentList << "-importPath" << importPath;
+
     QString qmlImportsPath = QLibraryInfo::path(QLibraryInfo::QmlImportsPath);
-    argumentList.append( "-importPath");
-    argumentList.append(qmlImportsPath);
+    if (QFile::exists(qmlImportsPath)) {
+        argumentList.append( "-importPath");
+        argumentList.append(qmlImportsPath);
+    }
 
     // In a modularized installation of qt as we have in Nix, instead, we will
     // read the paths from the environment, as they are spread in multiple
